### PR TITLE
[JSQMessagesViewController] to Nuget.

### PR DESCRIPTION
**Pull request by adlair**

### CODE REVIEW

File: iOS/JSQMessagesViewController/build.cake [hunk 1]
```diff
@@ -3,6 +3,14 @@
 
 var TARGET = Argument ("t", Argument ("target", "Default"));
 
+var IOS_PODS = new List<string> {
```
Comment (bholmes) [inline iOS/JSQMessagesViewController/build.cake:6]: Do not fetch/build the native library this way.  Take a look at the `externals/Makefile` . It fetches the code, makes some changes and builds the library locally.

File: iOS/JSQMessagesViewController/nuget/Xamarin.JSQMessagesViewController.nuspec [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  <metadata>
+    <id>Xamarin.JSQMessagesViewController</id>
+    <title>JSQMessagesViewController for Xamarin.iOS</title>
+    <version>1.11.7</version>
+    <authors>Microsoft</authors>
+    <owners>Microsoft</owners>
+    <requireLicenseAcceptance>true</requireLicenseAcceptance>
+    <description>An elegant messages UI library for iOS.</description>
+    <copyright>© Microsoft Corporation. All rights reserved.</copyright>
+    <projectUrl>https://components.xamarin.com/view/jsqmessagesviewcontroller</projectUrl>
```
Comment (bholmes) [inline iOS/JSQMessagesViewController/nuget/Xamarin.JSQMessagesViewController.nuspec]: Use FWLinks. Point to Github and move license.md file from components folder to root of the project (`iOS/JSQMessagesViewController`)

File: iOS/JSQMessagesViewController/build.cake [hunk 3]
```diff
@@ -3,6 +3,14 @@
...
+	"platform :ios, '11.2'",
```
Comment (SotoiGhost) [inline iOS/JSQMessagesViewController/build.cake]: Does this pod support only the most recent iOS version?

File: iOS/JSQMessagesViewController/component/component.yaml [hunk 4]
```diff
@@ -2,7 +2,7 @@
 ---
 name: JSQMessagesViewController
 id: jsqmessagesviewcontroller
-version: 7.3.4
+version: 7.3.5
```
Comment (newky2k) [inline iOS/JSQMessagesViewController/component/component.yaml]: This file needs to have the is_shell set and to reference the nuget that is specified in the spec.

Also need to move the licence.md file to the root folder from the components folder

This is an example -> https://github.com/newky2k/XamarinComponents/blob/dh_InAppSettingsKit/iOS/InAppSettingsKit/component/component.yaml#L22-L24

File: iOS/JSQMessagesViewController/nuget/Xamarin.JSQMessagesViewController.nuspec [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    <licenseUrl>https://components.xamarin.com/license/jsqmessagesviewcontroller</licenseUrl>
```
Comment (newky2k) [inline iOS/JSQMessagesViewController/nuget/Xamarin.JSQMessagesViewController.nuspec]: USE FWLinks as above

File: iOS/JSQMessagesViewController/nuget/Xamarin.JSQMessagesViewController.nuspec [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  </metadata>
+  <files>
+    <file src="output/unified/JSQMessagesViewController.dll" target="lib/Xamarin.iOS" />
+  </files>
```
Comment (newky2k) [inline iOS/JSQMessagesViewController/nuget/Xamarin.JSQMessagesViewController.nuspec]: Need to add the external dependecies file

File: iOS/JSQMessagesViewController/component/component.yaml [hunk 7]
```diff
@@ -2,7 +2,7 @@
 ---
 name: JSQMessagesViewController
 id: jsqmessagesviewcontroller
-version: 7.3.4
+version: 7.3.5
 summary: An elegant messages UI library for iOS
 publisher: Xamarin Inc.
```
Comment (newky2k) [inline iOS/JSQMessagesViewController/component/component.yaml]: No license link

**Review by newky2k — CHANGES_REQUESTED** [review]

File: iOS/JSQMessagesViewController/component/component.yaml [hunk 8]
```diff
@@ -18,6 +18,10 @@ libraries:
 samples: 
   - name: "Xamarin Chat Sample"
     path: ../samples/XamarinChat.sln
+    installNuGets:
```
Comment (newky2k) [inline iOS/JSQMessagesViewController/component/component.yaml]: Missing the remove part of the install nugets

File: iOS/JSQMessagesViewController/component/component.yaml [hunk 9]
```diff
@@ -18,6 +18,10 @@ libraries:
 samples: 
   - name: "Xamarin Chat Sample"
     path: ../samples/XamarinChat.sln
+    installNuGets:
+      - project: XamarinChat
+        packages:
+          - Xamarin.JSQMessagesViewController
     configuration: "Release|iPhone"
 no_build: true
```
Comment (newky2k) [inline iOS/JSQMessagesViewController/component/component.yaml:29]: No local-nuget-repo tag and not is_shell tag either

Comment (newky2k) [inline iOS/JSQMessagesViewController/component/component.yaml:29]: No externals file being included in the xam package

**Review by newky2k — CHANGES_REQUESTED** [review]

**Comment by newky2k** [comment]
Fails to compile when building the component -> https://gist.github.com/newky2k/deecfe36969f72b55f0696d5075abdd6

File: iOS/JSQMessagesViewController/nuget/Xamarin.JSQMessagesViewController.nuspec [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    <authors>Microsoft</authors>
+    <owners>Microsoft</owners>
+    <requireLicenseAcceptance>true</requireLicenseAcceptance>
+    <description>An elegant messages UI library for iOS.</description>
+    <copyright>© Microsoft Corporation. All rights reserved.</copyright>
+    <projectUrl>https://go.microsoft.com/fwlink/?linkid=869726</projectUrl>
+    <licenseUrl>https://go.microsoft.com/fwlink/?linkid=869724</licenseUrl>
+  </metadata>
+  <files>
+    <file src="output/unified/JSQMessagesViewController.dll" target="lib/Xamarin.iOS" />
```
Comment (newky2k) [inline iOS/JSQMessagesViewController/nuget/Xamarin.JSQMessagesViewController.nuspec:16]: The dll is not in unified in the output folder, just in the output folder itself

**Review by newky2k — CHANGES_REQUESTED** [review]

File: iOS/JSQMessagesViewController/component/component.yaml (hunk 9 above)
Comment (bholmes) [reply iOS/JSQMessagesViewController/component/component.yaml:29]: Do you mean the Third Party Notice?

**Review by newky2k — CHANGES_REQUESTED** [review]
The sample needs updating to support 64bit for the simulator.  Its currently just I386 which won't now run in iOS 11

**Comment by adlair** [comment]
I added the changes on https://github.com/xamarin/XamarinComponents/pull/358.

**Comment by bholmes** [comment]
There was a problem with Git and GitHub where the PR was disconnected from the fork.  This PR is patched and opened again in #358.

**Pull request closed without merging**